diff --git a/greeter-server/app.py b/greeter-server/app.py
index 37baafe..7ed534d 100644
--- a/greeter-server/app.py
+++ b/greeter-server/app.py
@@ -1,13 +1,14 @@
-from flask import Flask, jsonify
+from flask import Flask, jsonify, request
 
 
 app = Flask(__name__)
 
 
-@app.route('/')
+@app.route('/', methods = ['POST'])
 def index():
-    """Returns a simple hello world message."""
-    return jsonify({'message': 'Hello, world.'})
+    """Returns a simple hello world message, greeting the requested greetee."""
+    greetee = request.json['greetee']
+    return jsonify({'message': f'Hello, {greetee}.'})
 
 
 app.run()
